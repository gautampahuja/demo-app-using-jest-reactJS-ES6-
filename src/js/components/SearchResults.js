import React from 'react';

class SearchResults extends React.Component {
    constructor (props) {
        super(props);
        this.state = {
            selectedTab: null
        };
        this.carClick = this.carClick.bind(this);
    }
    carClick(name) {
        this.setState({
            selectedTab: name
        });
        this.props.onCarClick(name);
    }

    render() {
        const filteredCars = this.props.filteredCars;
        const self = this;
        return (
            <div id="search-Results" className="results">
                <ul id="cars-list">
                    {
                        filteredCars!=undefined
                            ?
                            filteredCars.map((obj, i) => <li key={i} className={self.state.selectedTab == obj.name ? 'active' : ''} onClick={self.carClick.bind(null, obj.name)}>{obj.name}</li>)
                            :
                            ""
                        }
                </ul>
            </div>
        )
    }
}
export default SearchResults;