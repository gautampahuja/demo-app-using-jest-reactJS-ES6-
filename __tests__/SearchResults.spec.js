/* global describe, test, expect */

import React from 'react'
import SearchResults from '../src/js/components/SearchResults.js'
import renderer from 'react-test-renderer'

describe('SearchResults component', () => {
  test('SearchResults list is rendered without filteredCars object', () => {
    const component = renderer.create(<SearchResults/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot()
  });

  test('SearchResults list is rendered with filteredCars object ', () => {
    const filteredCars = [{
        "name": "MERCEDES BENZ"
    }, {
        "name": "VOLKS WAGEN"
    },{
        "name": "HYUNDAI"
    },{
        "name": "INFINITI"
    }];
    const component = renderer.create(<SearchResults filteredCars={filteredCars} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot()
  })
})
