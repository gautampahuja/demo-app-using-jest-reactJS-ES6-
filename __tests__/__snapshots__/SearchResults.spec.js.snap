// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults component SearchResults list is rendered with filteredCars object  1`] = `
<div
  className="results"
  id="search-Results"
>
  <ul
    id="cars-list"
  >
    <li
      className=""
      onClick={[Function]}
    >
      MERCEDES BENZ
    </li>
    <li
      className=""
      onClick={[Function]}
    >
      VOLKS WAGEN
    </li>
    <li
      className=""
      onClick={[Function]}
    >
      HYUNDAI
    </li>
    <li
      className=""
      onClick={[Function]}
    >
      INFINITI
    </li>
  </ul>
</div>
`;

exports[`SearchResults component SearchResults list is rendered without filteredCars object 1`] = `
<div
  className="results"
  id="search-Results"
>
  <ul
    id="cars-list"
  >
    
  </ul>
</div>
`;
